modules/evm-manager/src/lib.rs

```diff
@@ -175,9 +175,8 @@ impl<T: Config> CurrencyIdMapping for EvmCurrencyIdMapping<T> {
 	fn encode_currency_id(val: CurrencyId) -> Option<[u8; 32]> {
 		let mut bytes = [0u8; 32];
 		match val {
-			CurrencyId::Token(_) => {
-				let id: u32 = val.try_into().expect("CurrencyId::Token into u32 is success; qed");
-				bytes[12..16].copy_from_slice(&id.to_be_bytes()[..]);
+			CurrencyId::Token(token) => {
+				bytes[31] = token as u8;
```

Comment:
`as` usage looks dangerous. I would prefer to have `Into<u8>` impl for `TokenSymbol` and call it, to ensure type safety.